### DISCUSSION TITLE
Planning Theater for WordPress 1.0

### DISCUSSION THREAD

**Discussion by slimndap** [slimndap/wp-theatre#321] [Announcements]
I am targeting a new releases of the Theater for WordPress plugin in early 2024, mainly sparked by [this issue](https://github.com/slimndap/wp-theatre/issues/316) by @carstingaxion.

Before defining the full scope of the release, I would like to better specify the main role of the plugin. Is the Theater for WordPress:

1. a base plugin that can be extended with plugins and hooks or
2. a fully featured events plugin?

## Base plugin

The main difference between Theater for WordPress and other popular calendar plugins, is the option to group multiple events within a single production. This makes the plugin especially suitable for theaters and cinemas with multiple performances by the same group or multiple screenings of the same movie. The plugin is also popular with music venues, in which case just productions with a single event are used.

The base plugin will facilitate the infrastructure and interface to manage and publish productions and events on a WordPress website. It can create and manage a fully functional events website for your theater, including listings and event detail pages. 

The front end output is very clean, to ensure accessibility and compatibility with themes and can be customized using hooks and template files.

The back end is very basic with very little settings. It just works. Everything follows WordPress guidelines to avoid conflicts with other plugins. More advanced features can be added through extensions, or custom code using hooks. 

## Fully featured plugin

All of the above, plus:

* Advanced filtering in listings.
* Multiple views in listings (eg. timetables, calendars, thumbnails, sliders).
* Ticket sales.
* A lot of settings.
* Gutenberg blocks.
* SEO.
* ....

## Current idea

I have a strong preference for the base plugin. It is easier to manage and a lot more flexible for developers and implementers. The current version of the plugin is not that far off, but could lose some features here and there.

**Comment by carstingaxion** [comment]
> I have a strong preference for the base plugin. 

This is absolutely my position as well. As I wrote #316 I had in mind to refactor the existing codebase to a more focused, but extensible base-plugin. 

I think, most of the existing features could be wrapped in a kind of extension template, that we create, to provide a good-starting point for others and ourself says well. This should help updating from previous versions, when feature-pairity is reached by just installing some *extension plugins*.

**Comment by carstingaxion** [comment]
When we can separate things nicely I can imagine a very little base plugin, which only handles to absolute minimum (what ever this means). And a kind-of repository with all the extensions. 

On the plugins website, you as a user, can select all necessary extensions you want and download a fresh bundle, which you can than **install in one go**.

The version hosted on WordPress would also be a bundle of multiple extensions to address backwards-compatibility with plugin-installations older than this release. The bundle that gets (automatically) deployed to svn.wordpress.org will contain everything, we're going to move into separate repositories within this 1.0 release.
